debezium-core/src/main/java/io/debezium/util/LoggingContext.java

```diff
@@ -62,22 +65,41 @@ public void restore() {
      * @param connectorName the name of the connector; may not be null
      * @param contextName the name of the context; may not be null
      * @return the previous MDC context; never null
-     * @throws IllegalArgumentException if any of the parameters are null
+     * @throws NullPointerException if any of the parameters are null
      */
     public static PreviousContext forConnector(String connectorType, String connectorName, String contextName) {
-        if (connectorType == null) {
-            throw new IllegalArgumentException("The MDC value for the connector type may not be null");
-        }
-        if (connectorName == null) {
-            throw new IllegalArgumentException("The MDC value for the connector name may not be null");
+        return forConnector(connectorType, connectorName, contextName, null, null);
+    }
+
+    /**
+     * Configure for a connector the logger's Mapped Diagnostic Context (MDC) properties for the thread making this call.
+     *
+     * @param connectorType the type of connector; may not be null
+     * @param connectorName the name of the connector; may not be null
+     * @param contextName the name of the context; may not be null
+     * @param taskId the task id; may be null
+     * @param partition the partition; may be null
+    
+     * @return the previous MDC context; never null
+     * @throws NullPointerException if connectorType, connectorName, or contextName parameters are null
+     */
+    public static PreviousContext forConnector(String connectorType, String connectorName, String contextName, String taskId, String partition) {
```

Comment:
Please reorder arguments and use `Partition partition`:
```suggestion
    public static PreviousContext forConnector(String connectorType, String connectorName, String taskId, String contextName, Partition partition) {
```